### PR TITLE
[Android] Add initial support of the platform

### PR DESCRIPTION
**Pull request by ghuau-innopeak**
Related issue = #3679

This is an initial support so a lot of features are not supported yet such as the logging with logcat or traceback.

### CODE REVIEW

**Comment by ghuau-innopeak** [comment]
I added a new tag [Android] as this is a new platform, but I can rename it to [Linux] if needed. Also, for testing, this is what I'm currently using:
```
TAICHI_CMAKE_ARGS="-DCMAKE_TOOLCHAIN_FILE=$ANDROID_HOME/ndk-bundle/build/cmake/android.toolchain.cmake -DANDROID_NATIVE_API_LEVEL=29 -DANDROID_ABI=arm64-v8a -DTI_WITH_VULKAN:BOOL=ON -DTI_WITH_CUDA:BOOL=OFF -DTI_WITH_OPENGL:BOOL=OFF -DTI_WITH_LLVM:BOOL=OFF -DTI_EXPORT_CORE:BOOL=ON" python3 setup.py build_ext
```

File: taichi/backends/vulkan/vulkan_program.cpp [hunk 1]
```diff
@@ -68,6 +85,9 @@ void VulkanProgramImpl::materialize_runtime(MemoryPool *memory_pool,
   *result_buffer_ptr = (uint64 *)memory_pool->allocate(
       sizeof(uint64) * taichi_result_buffer_entries, 8);
 
+// TODO: Support for Android if possible... Should we move this code to be the
+// responsability of the caller to this library?
```
Comment (bobcao3) [inline taichi/backends/vulkan/vulkan_program.cpp:89]: When embedded in other applications the design is to leave the creation of VkDevice and related states to the caller/host. VulkanDeviceCreator is only used when Taichi is running on its own.

**Review by bobcao3 — APPROVED** [review]
Looks great!

File: taichi/backends/vulkan/vulkan_program.cpp (hunk 1 above)
Comment (bobcao3) [reply taichi/backends/vulkan/vulkan_program.cpp]: I think it's good for now!

**Pull request merged**